React/creating-components.md

```diff
@@ -0,0 +1,79 @@
+
+### Overview
+It is important to understand what is happening currently within the React application so we can model the same pattern with a custom component. Right now, our React application renders the `App` component from the `index.js` file and puts it in the DOM element where the ID is "root". Normally, this is only done **once** in a React application, so we are going to leave this code alone as we build on.
+
+Next, our `App` component contains a render function will which render a `header` and a `p` using JSX.
+
+The powerful thing about React is that **any component can render any other component**. That is, we can (and should) **nest our components** within one another to create the most effective React application design.
+
+### Context
+As we continue using this application, we are going to be creating a React application that manages our list of Ada students! We are going to use different components to handle different portions of our student listing.
+
+### Create a new Component
+React does not have a nice CLI interface like Rails to create new components for us. We will be creating these manually by creating new files, importing the necessary libraries, and adding the necessary component code. We'll use the `Component` class provided by React and create a class that extends this class in order to leverage this functionality.
+
+We are going to start small, and our first component will be used to display a single student's information.
+
+1. Create a new folder called `components` where we'll store all of our custom components. This should be under the existing `src` directory.
+
+1. Create a new file in this new folder called `student.js`
+
+1. In this new file, import the React and Component libraries:
+    ```javascript
+    import React, { Component } from 'react';
+    ```
+
+1. Create the class definition with the name of the component you want to create. Note that this does not need to match the file name.
+    ```javascript
+    class Student extends Component {
+
+    }
+    ```
+
+1. Export the class at the bottom of the file. Note that whatever is exported here needs to match the name of the class created in the step above. This will allow us to import this component to use in other files.
+    ```JavaScript
+    export default Student;
+    ```
+
+1. **QUICK BREAK!** Check in with your seat squad to verify that they are at the same place in setting up their component file. Look at [the code](https://github.com/AdaGold/react-hello-world/blob/part-1/src/components/student.js) if you need to copy/paste to catch up.
+
+### Customize our Component
+
+With your seat squad, add some JSX elements to a `render()` function that you add to this new component. We can't fully test it out quite yet, but you can make sure that your React application compiles and does not show you any errors in the browser as confirmation of success. Feel free to hard-code some student data and HTML that you think might be relevant.
```

Comment:
I'm certainly curious to see how this turns out. Up until this point the students have had very little opportunity to try writing their own JSX or `render()` function...

Also, until they have an `import` line for the new Student component's file, React will not attempt to compile this file and won't give any errors if there are syntax issues. Should we briefly go over the terminal output from `npm start` (or the browser error display) and how to identify when there are compilation errors?